webproject/src/main/java/com/google/sps/servlets/UsernameServlet.java

```diff
@@ -26,7 +36,10 @@ public UsernameServlet() {
      * @throws IOException If an input or output error is detected when the servlet handles the server's request.
      */
     @Override
-    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        response.getWriter().print(authenticationHandler.getCurrentUser().getNickname());
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
+        UsernameRequest usernameRequest = UsernameRequest.newBuilder().build();
+        UsernameResponse usernameResponse = usernameService.execute(usernameRequest);
+
+        response.getWriter().print(usernameResponse.getUsername());
```

Comment:
return as json of UsernameResponse

Reply:
this way in the future in case u need to add more data there wont be a lot of refactor